### PR TITLE
Fixes for StressLog handling

### PR DESCRIPTION
**Pull request by leculver**
This pull request enhances the stress log integration and testing infrastructure by exposing additional configuration fields for in-process logs, improving memory range enumeration robustness, and expanding test coverage to validate these behaviors. The changes affect both the `StressLog` implementation and its associated tests, ensuring more accurate diagnostics and safer handling of edge cases.

**Enhancements to StressLog configuration and API:**

- Exposed new properties on the `StressLog` class for in-process logs: `FacilitiesToLog`, `LevelToLog`, `MaxSizePerThread`, `MaxSizeTotal`, and `ChunkCount`, all nullable and unavailable for memory-mapped logs. These are now parsed from the log header and surfaced as public properties. [[1]](diffhunk://#diff-4eb448cff27aca71a5ad6ca9b12a7a049fdb0468af20d3c3862b661725a0255aR52-R56) [[2]](diffhunk://#diff-4eb448cff27aca71a5ad6ca9b12a7a049fdb0468af20d3c3862b661725a0255aL65-R75) [[3]](diffhunk://#diff-4eb448cff27aca71a5ad6ca9b12a7a049fdb0468af20d3c3862b661725a0255aR88-R92) [[4]](diffhunk://#diff-4eb448cff27aca71a5ad6ca9b12a7a049fdb0468af20d3c3862b661725a0255aR118-R156) [[5]](diffhunk://#diff-4eb448cff27aca71a5ad6ca9b12a7a049fdb0468af20d3c3862b661725a0255aR306-R310) [[6]](diffhunk://#diff-4eb448cff27aca71a5ad6ca9b12a7a049fdb0468af20d3c3862b661725a0255aL292-R355)
- Made `StressLog.TryOpen(IDataReader, ulong, ...)` public to allow callers to open logs at arbitrary addresses, not just the runtime's default.

**Expanded and improved test coverage:**

- Added comprehensive tests in `StressLogTests` to verify:
  - The new header config fields are correctly exposed for in-process logs and are null for memory-mapped logs.
  - The `EnumerateMemoryRanges` method covers all live chunk memory, handles thread and chunk cycles gracefully, supports multiple threads, and continues enumeration even when encountering corrupt threads.
- Updated `StressLogIntegrationTests` to assert the presence and correctness of the new header config fields and to validate memory range enumeration for live chunk memory.

**Supporting changes:**

- Added `System.Buffers.Binary` import to support low-level header parsing in tests.

### CODE REVIEW

File: src/Microsoft.Diagnostics.Runtime/StressLog/StressLog.cs [hunk 1]
```diff
@@ -318,9 +381,138 @@ internal static bool TryOpen(IDataReader reader, ulong address, out StressLog? s
...
+                            }
+                            if (!_budget.TryReserve(visitedChunkBytes))
+                            {
+                                // Global memory ceiling: stop the whole enumeration.
+                                RaiseDiagnostic(StressLogDiagnosticKind.LimitExceeded, chunkAddr);
+                                yield break;
+                            }
+                            reservedBytes += visitedChunkBytes;
+
+                            yield return new MemoryRange(chunkAddr, chunkAddr + (ulong)_layout.ChunkTotalSize);
```
Comment (max-charlamb) [inline src/Microsoft.Diagnostics.Runtime/StressLog/StressLog.cs]: It looks like we yield the memory range before we validate it is readable. Based on the docs:
`A thread or chunk that cannot be read, or that looks corrupt, is skipped and reported through Diagnostic; enumeration continues with the remaining threads where possible.`

It looks like this shouldn't be yielded.

Comment (leculver) [reply src/Microsoft.Diagnostics.Runtime/StressLog/StressLog.cs]: Good catch, thanks.

**Review by max-charlamb — APPROVED** [review]
